src/ralph/api/auth/user.py

```diff
@@ -18,6 +19,42 @@
 ]
 
 
+class UserScopes(FrozenSet[Scope]):
+    """Scopes available to users."""
+
+    @lru_cache()
+    def is_authorized(self, requested_scope: Scope):
+        """Check if the requested scope can be accessed based on user scopes."""
+        expanded_scopes = {
+            "statements/read": {"statements/read/mine", "statements/read"},
+            "all/read": {
+                "statements/read/mine",
+                "statements/read",
+                "state/read",
+                "profile/read",
+                "all/read",
+            },
+            "all": {
+                "statements/write",
+                "statements/read/mine",
+                "statements/read",
+                "state/read",
+                "state/write",
+                "define",
+                "profile/read",
+                "profile/write",
+                "all/read",
+                "all",
+            },
+        }
+
+        expanded_user_scopes = set()
+        for scope in self:
+            expanded_user_scopes.update(expanded_scopes.get(scope, {scope}))
```

Comment:
Yes, but I don't see the use for this as required scopes in an endpoint are always "leafs" of the tree.

Reply:
hmm yes I see what you mean,  hadn't thought of that.

But I feel like even in the event of scopes not being "leafs" (for example with a scope `"read/all"`), it would be preferable not to infer that `"statements/read + state/read +  profile/read" <=> "read/all"`, to not grant access to protected functions by mistake.

Reply:
Good point) Ok, thus, we consider the `read/all` scope to be more than just an alias for `{statements,state,profile}/read` scopes. It's clearer to me now) Thanks)